src/UKHO.S100PermitService.Common/Services/ProductKeyService.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System.Net;
+using UKHO.S100PermitService.Common.Clients;
+using UKHO.S100PermitService.Common.Configuration;
+using UKHO.S100PermitService.Common.Events;
+using UKHO.S100PermitService.Common.Exceptions;
+using UKHO.S100PermitService.Common.Models.ProductKeyService;
+using UKHO.S100PermitService.Common.Providers;
+
+namespace UKHO.S100PermitService.Common.Services
+{
+    public class ProductKeyService : IProductKeyService
+    {
+        private readonly ILogger<ProductKeyService> _logger;
+        private readonly IOptions<ProductKeyServiceApiConfiguration> _productKeyServiceApiConfiguration;
+        private readonly IProductKeyServiceAuthTokenProvider _productKeyServiceAuthTokenProvider;
+        private readonly IProductKeyServiceApiClient _productKeyServiceApiClient;
+        private const string KeysEnc = "/keys/s100";
+
+        public ProductKeyService(ILogger<ProductKeyService> logger, IOptions<ProductKeyServiceApiConfiguration> productKeyServiceApiConfiguration, IProductKeyServiceAuthTokenProvider productKeyServiceAuthTokenProvider, IProductKeyServiceApiClient productKeyServiceApiClient)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _productKeyServiceApiConfiguration = productKeyServiceApiConfiguration ?? throw new ArgumentNullException(nameof(productKeyServiceApiConfiguration));
+            _productKeyServiceAuthTokenProvider = productKeyServiceAuthTokenProvider ?? throw new ArgumentNullException(nameof(productKeyServiceAuthTokenProvider));
+            _productKeyServiceApiClient = productKeyServiceApiClient ?? throw new ArgumentNullException(nameof(productKeyServiceApiClient));
+        }
+
+        /// <summary>
+        /// Get permit key from Product Key Service
+        /// </summary>
+        /// <param name="productKeyServiceRequest"></param>
+        /// <param name="correlationId"></param>
+        /// <returns>ProductKeyServiceResponse</returns>
+        /// <exception cref="Exception"></exception>
+        public async Task<List<ProductKeyServiceResponse>> PostProductKeyServiceRequestAsync(List<ProductKeyServiceRequest> productKeyServiceRequest, string correlationId)
```

Comment:
The method name should tell a consumer what a method does, not how it does it - so should be something like GetPermitKeys(...). It should probably create the ProductKeyServiceRequest from parameters, not have it passed in, this breaks encapsulation.